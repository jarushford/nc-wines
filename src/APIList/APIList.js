import React from 'react'

export default function APIList() {
  return (
    <section className="api-list">
      <h2>All Endpoints</h2>
      
    </section>
  )
}