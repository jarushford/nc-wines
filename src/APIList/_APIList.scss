.api-list {
  height: 200vh;
}