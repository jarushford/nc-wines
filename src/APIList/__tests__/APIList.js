import React from 'react'
import { shallow } from 'enzyme'
import APIList from '../APIList'

describe('APIList', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<APIList />)
    expect(wrapper).toMatchSnapshot()
  })
})