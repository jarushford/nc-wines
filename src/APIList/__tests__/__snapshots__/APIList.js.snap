// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APIList should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <APIList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Available Endpoints
        </h2>,
        <Endpoint
          description="Returns an array of all vineyards. This endpoint can also be modified with 'name' and 'region' query parameters to search more specifically."
          method="GET"
          required="N/A"
          returned="Array"
          title="/api/v1/vineyards"
        />,
        <Endpoint
          description="Returns a single vineyard."
          method="GET"
          required="N/A"
          returned="Object"
          title="/api/v1/vineyards/:id"
        />,
        <Endpoint
          description="Creates a new vineyard."
          method="POST"
          required="{ name: <String>, address: <String>, website: <String>, phone: <String>, region: <String> }"
          returned="Object with ID of new vineyard"
          title="/api/v1/vineyards/"
        />,
        <Endpoint
          description="Edits an exisiting vineyard."
          method="PUT"
          required="Object with the parameters you wish to update"
          returned="Object with ID of updated vineyard"
          title="/api/v1/vineyards/:id"
        />,
        <Endpoint
          description="Deletes an existing vineyard."
          method="DELETE"
          required="N/A"
          returned="Object with ID of deleted vineyard"
          title="/api/v1/vineyards/:id"
        />,
        <Endpoint
          description="Returns an array of all wines."
          method="GET"
          required="N/A"
          returned="Array"
          title="/api/v1/wines"
        />,
        <Endpoint
          description="Returns a single wine."
          method="GET"
          required="N/A"
          returned="Object"
          title="/api/v1/wines/:id"
        />,
        <Endpoint
          description="Creates a new wine."
          method="POST"
          required="{ name: <String>, type: <String>, color: <String>, vineyard_id: <Number> }"
          returned="Object with ID of new wine"
          title="/api/v1/wines"
        />,
        <Endpoint
          description="Edits an existing wine."
          method="PUT"
          required="Object with the parameters you wish to update"
          returned="Object with ID of updated wine"
          title="/api/v1/wines/:id"
        />,
        <Endpoint
          description="Deletes an existing wine."
          method="DELETE"
          required="N/A"
          returned="Object with ID of deleted wine"
          title="/api/v1/wines/:id"
        />,
      ],
      "className": "api-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Available Endpoints",
        },
        "ref": null,
        "rendered": "Available Endpoints",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Returns an array of all vineyards. This endpoint can also be modified with 'name' and 'region' query parameters to search more specifically.",
          "method": "GET",
          "required": "N/A",
          "returned": "Array",
          "title": "/api/v1/vineyards",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Returns a single vineyard.",
          "method": "GET",
          "required": "N/A",
          "returned": "Object",
          "title": "/api/v1/vineyards/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Creates a new vineyard.",
          "method": "POST",
          "required": "{ name: <String>, address: <String>, website: <String>, phone: <String>, region: <String> }",
          "returned": "Object with ID of new vineyard",
          "title": "/api/v1/vineyards/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Edits an exisiting vineyard.",
          "method": "PUT",
          "required": "Object with the parameters you wish to update",
          "returned": "Object with ID of updated vineyard",
          "title": "/api/v1/vineyards/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Deletes an existing vineyard.",
          "method": "DELETE",
          "required": "N/A",
          "returned": "Object with ID of deleted vineyard",
          "title": "/api/v1/vineyards/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Returns an array of all wines.",
          "method": "GET",
          "required": "N/A",
          "returned": "Array",
          "title": "/api/v1/wines",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Returns a single wine.",
          "method": "GET",
          "required": "N/A",
          "returned": "Object",
          "title": "/api/v1/wines/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Creates a new wine.",
          "method": "POST",
          "required": "{ name: <String>, type: <String>, color: <String>, vineyard_id: <Number> }",
          "returned": "Object with ID of new wine",
          "title": "/api/v1/wines",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Edits an existing wine.",
          "method": "PUT",
          "required": "Object with the parameters you wish to update",
          "returned": "Object with ID of updated wine",
          "title": "/api/v1/wines/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Deletes an existing wine.",
          "method": "DELETE",
          "required": "N/A",
          "returned": "Object with ID of deleted wine",
          "title": "/api/v1/wines/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Available Endpoints
          </h2>,
          <Endpoint
            description="Returns an array of all vineyards. This endpoint can also be modified with 'name' and 'region' query parameters to search more specifically."
            method="GET"
            required="N/A"
            returned="Array"
            title="/api/v1/vineyards"
          />,
          <Endpoint
            description="Returns a single vineyard."
            method="GET"
            required="N/A"
            returned="Object"
            title="/api/v1/vineyards/:id"
          />,
          <Endpoint
            description="Creates a new vineyard."
            method="POST"
            required="{ name: <String>, address: <String>, website: <String>, phone: <String>, region: <String> }"
            returned="Object with ID of new vineyard"
            title="/api/v1/vineyards/"
          />,
          <Endpoint
            description="Edits an exisiting vineyard."
            method="PUT"
            required="Object with the parameters you wish to update"
            returned="Object with ID of updated vineyard"
            title="/api/v1/vineyards/:id"
          />,
          <Endpoint
            description="Deletes an existing vineyard."
            method="DELETE"
            required="N/A"
            returned="Object with ID of deleted vineyard"
            title="/api/v1/vineyards/:id"
          />,
          <Endpoint
            description="Returns an array of all wines."
            method="GET"
            required="N/A"
            returned="Array"
            title="/api/v1/wines"
          />,
          <Endpoint
            description="Returns a single wine."
            method="GET"
            required="N/A"
            returned="Object"
            title="/api/v1/wines/:id"
          />,
          <Endpoint
            description="Creates a new wine."
            method="POST"
            required="{ name: <String>, type: <String>, color: <String>, vineyard_id: <Number> }"
            returned="Object with ID of new wine"
            title="/api/v1/wines"
          />,
          <Endpoint
            description="Edits an existing wine."
            method="PUT"
            required="Object with the parameters you wish to update"
            returned="Object with ID of updated wine"
            title="/api/v1/wines/:id"
          />,
          <Endpoint
            description="Deletes an existing wine."
            method="DELETE"
            required="N/A"
            returned="Object with ID of deleted wine"
            title="/api/v1/wines/:id"
          />,
        ],
        "className": "api-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Available Endpoints",
          },
          "ref": null,
          "rendered": "Available Endpoints",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Returns an array of all vineyards. This endpoint can also be modified with 'name' and 'region' query parameters to search more specifically.",
            "method": "GET",
            "required": "N/A",
            "returned": "Array",
            "title": "/api/v1/vineyards",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Returns a single vineyard.",
            "method": "GET",
            "required": "N/A",
            "returned": "Object",
            "title": "/api/v1/vineyards/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Creates a new vineyard.",
            "method": "POST",
            "required": "{ name: <String>, address: <String>, website: <String>, phone: <String>, region: <String> }",
            "returned": "Object with ID of new vineyard",
            "title": "/api/v1/vineyards/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Edits an exisiting vineyard.",
            "method": "PUT",
            "required": "Object with the parameters you wish to update",
            "returned": "Object with ID of updated vineyard",
            "title": "/api/v1/vineyards/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Deletes an existing vineyard.",
            "method": "DELETE",
            "required": "N/A",
            "returned": "Object with ID of deleted vineyard",
            "title": "/api/v1/vineyards/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Returns an array of all wines.",
            "method": "GET",
            "required": "N/A",
            "returned": "Array",
            "title": "/api/v1/wines",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Returns a single wine.",
            "method": "GET",
            "required": "N/A",
            "returned": "Object",
            "title": "/api/v1/wines/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Creates a new wine.",
            "method": "POST",
            "required": "{ name: <String>, type: <String>, color: <String>, vineyard_id: <Number> }",
            "returned": "Object with ID of new wine",
            "title": "/api/v1/wines",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Edits an existing wine.",
            "method": "PUT",
            "required": "Object with the parameters you wish to update",
            "returned": "Object with ID of updated wine",
            "title": "/api/v1/wines/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Deletes an existing wine.",
            "method": "DELETE",
            "required": "N/A",
            "returned": "Object with ID of deleted wine",
            "title": "/api/v1/wines/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
